exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestCastFunctions.java

```diff
@@ -752,18 +770,59 @@ public void testCastTimeLiteralInFilter() throws Exception {
 
   @Test
   public void testCastUntypedNull() throws Exception {
-    String[] types = new String[] {
-        "BOOLEAN", "INT", "BIGINT", "FLOAT", "DOUBLE", "DATE", "TIME", "TIMESTAMP", "INTERVAL MONTH",
-        "INTERVAL YEAR", "VARBINARY", "VARCHAR", "DECIMAL(9)", "DECIMAL(18)", "DECIMAL(28)", "DECIMAL(38)"
-    };
     String query = "select cast(coalesce(unk1, unk2) as %s) as coal from cp.`tpch/nation.parquet` limit 1";
-    for (String type : types) {
+
+    Map<String, Triple<TypeProtos.MinorType, Integer, Integer>> typesMap = createCastTypeMap();
+    for (Map.Entry<String, Triple<TypeProtos.MinorType, Integer, Integer>> entry : typesMap.entrySet()) {
+      String q = String.format(query, entry.getKey());
+
+      TypeProtos.MinorType minorType = entry.getValue().getLeft();
+      int precision = entry.getValue().getMiddle();
+      int scale = entry.getValue().getRight();
+      MaterializedField field = new ColumnBuilder("coal", minorType)
+          .setMode(TypeProtos.DataMode.OPTIONAL)
+          .setPrecisionAndScale(precision, scale)
+          .build();
+      BatchSchema expectedSchema = new SchemaBuilder()
+          .add(field)
+          .build();
+
+      // Validate schema
+      testBuilder()
+          .sqlQuery(q)
+          .schemaBaseLine(expectedSchema)
+          .go();
+
+      // Validate result
       testBuilder()
-          .sqlQuery(String.format(query, type))
+          .sqlQuery(q)
           .unOrdered()
           .baselineColumns("coal")
           .baselineValues(new Object[] {null})
           .go();
     }
   }
+
+  private static Map<String, Triple<TypeProtos.MinorType, Integer, Integer>> createCastTypeMap() {
+    Map<String, Triple<TypeProtos.MinorType, Integer, Integer>> typesMap = new HashMap<>();
```

Comment:
It would be better to pass `MaterializedField` instance instead of passing `TypeProtos.MinorType, Integer, Integer`.

Reply:
Good point. made `Map<String, TypeProtos.MajorType>` instead.